server/src/main/java/org/opensearch/gateway/remote/model/RemoteClusterBlocks.java

```diff
@@ -0,0 +1,102 @@
+import java.io.InputStream;
+import java.util.List;
+
+import static org.opensearch.core.common.bytes.BytesReference.toBytes;
+import static org.opensearch.gateway.remote.RemoteClusterStateAttributesManager.CLUSTER_STATE_ATTRIBUTES_CURRENT_CODEC_VERSION;
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.CLUSTER_STATE_EPHEMERAL_PATH_TOKEN;
+import static org.opensearch.gateway.remote.RemoteClusterStateUtils.DELIMITER;
+
+/**
+ * Wrapper class for uploading/downloading {@link ClusterBlocks} to/from remote blob store
+ */
+public class RemoteClusterBlocks extends AbstractRemoteWritableBlobEntity<ClusterBlocks> {
+
+    public static final String CLUSTER_BLOCKS = "blocks";
+
+    private ClusterBlocks clusterBlocks;
+    private long stateVersion;
+
+    public RemoteClusterBlocks(
+        final ClusterBlocks clusterBlocks,
+        long stateVersion,
+        String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry
+    ) {
+        super(clusterUUID, compressor, namedXContentRegistry);
+        this.clusterBlocks = clusterBlocks;
+        this.stateVersion = stateVersion;
+    }
+
+    public RemoteClusterBlocks(
+        final String blobName,
+        final String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry
+    ) {
+        super(clusterUUID, compressor, namedXContentRegistry);
+        this.blobName = blobName;
+    }
+
+    @Override
+    public BlobPathParameters getBlobPathParameters() {
+        return new BlobPathParameters(List.of(CLUSTER_STATE_EPHEMERAL_PATH_TOKEN), CLUSTER_BLOCKS);
+    }
+
+    @Override
+    public String generateBlobFileName() {
+        // 123456789012_test-cluster/cluster-state/dsgYj10Nkso7/transient/<componentPrefix>__<inverted_state_version>__<inverted__timestamp>__<codec_version>
+        String blobFileName = String.join(
+            DELIMITER,
+            getBlobPathParameters().getFilePrefix(),
+            RemoteStoreUtils.invertLong(stateVersion),
+            RemoteStoreUtils.invertLong(System.currentTimeMillis()),
+            String.valueOf(CLUSTER_STATE_ATTRIBUTES_CURRENT_CODEC_VERSION)
+        );
+        this.blobFileName = blobFileName;
+        return blobFileName;
+    }
+
+    @Override
+    public UploadedMetadata getUploadedMetadata() {
+        assert blobName != null;
+        return new UploadedMetadataAttribute(CLUSTER_BLOCKS, blobName);
+    }
+
+    @Override
+    public InputStream serialize() throws IOException {
+        BytesStreamOutput bytesStreamOutput = new BytesStreamOutput();
+        clusterBlocks.writeTo(bytesStreamOutput);
+        return bytesStreamOutput.bytes().streamInput();
+    }
+
+    @Override
+    public ClusterBlocks deserialize(final InputStream inputStream) throws IOException {
+        StreamInput in = new BytesStreamInput(toBytes(Streams.readFully(inputStream)));
+        return ClusterBlocks.readFrom(in);
+    }
```

Comment:
enclose ByteStreamOutput and Input within try-close

Reply:
+1, applies to other entities classes as well.